website/src/components/Tasks/Task.tsx

```diff
@@ -0,0 +1,76 @@
+import { useColorMode } from "@chakra-ui/react";
+import Head from "next/head";
+import { useEffect, useState } from "react";
+import { LoadingScreen } from "src/components/Loading/LoadingScreen";
+import fetcher from "src/lib/fetcher";
+import poster from "src/lib/poster";
+import useSWRImmutable from "swr/immutable";
+import useSWRMutation from "swr/mutation";
+
+import CreateTask from "./CreateTask";
+import EvaluateTask from "./EvaluateTask";
+import { taskTypes } from "./TaskTypes";
+
+const Task = ({ taskTypeName, taskName }) => {
+  const taskType = taskTypes[taskTypeName][taskName];
+  const [tasks, setTasks] = useState([]);
+
+  const { isLoading, mutate } = useSWRImmutable(`/api/new_task/${taskType.endpoint}`, fetcher, {
```

Comment:
The one big change I think is that this initial data fetch should be at the page level.  

That'll make the second PR easier.  The specific task pages would just fetch their known task type.  That code would be duplicated a bit but oh well.  The random task would fetch data from `/api/new_task/random` and then have a switch based on what task type gets returned.  It'd be able to render any of the available task types dynamically.  And then when we re-fetch a task, it should naturally just switch between different types.

Reply:
Thanks a lot for the feedback. I'm closing this PR and will open a new one once I have something to show.